### PR TITLE
DROOLS-2662: Prevent NPE when no dropDownData available

### PR DESCRIPTION
**Pull request by jomarko**
For more details, see the issue [DROOLS-2662](https://issues.jboss.org/browse/DROOLS-2662)

@manstis @Rikkola I will be pleased with your review, especially I am interested if the added tests reproduce/verify the issue enough.

### CODE REVIEW

File: drools-wb-screens/drools-wb-guided-rule-editor/drools-wb-guided-rule-editor-client/src/main/java/org/drools/workbench/screens/guided/rule/client/editor/util/ConstraintValueEditorHelper.java [hunk 1]
```diff
@@ -239,7 +240,12 @@ private void isLHSFieldTypeEquivalent(final String boundVariable,
     }
 
     public static boolean isEnumEquivalent(final String[] values, final DropDownData dropDownData) {
-        return Arrays.equals(values, dropDownData.getFixedList());
```
Comment (jomarko) [inline drools-wb-screens/drools-wb-guided-rule-editor/drools-wb-guided-rule-editor-client/src/main/java/org/drools/workbench/screens/guided/rule/client/editor/util/ConstraintValueEditorHelper.java]: This was the root cause, the `dropDownData` was null.

**Comment by jomarko** [comment]
jenkins execute full downstream build

File: drools-wb-screens/drools-wb-guided-rule-editor/drools-wb-guided-rule-editor-client/src/main/java/org/drools/workbench/screens/guided/rule/client/editor/util/ConstraintValueEditorHelper.java (hunk 1 above)
Comment (manstis) [reply drools-wb-screens/drools-wb-guided-rule-editor/drools-wb-guided-rule-editor-client/src/main/java/org/drools/workbench/screens/guided/rule/client/editor/util/ConstraintValueEditorHelper.java]: @jomarko Do you know why the ```dropDownData``` is ```null``` for ```java.time.LocalDate``` but not, for example, when ```java.util.Date``` (the JIRA states the issue is only evident when using ```LocalDate```)? I have no doubt this PR fixes the symptom but I'd like to know why ```dropDownData``` is ```null``` in the first place.

Comment (manstis) [reply drools-wb-screens/drools-wb-guided-rule-editor/drools-wb-guided-rule-editor-client/src/main/java/org/drools/workbench/screens/guided/rule/client/editor/util/ConstraintValueEditorHelper.java]: @jomarko I see ```ConstraintValueEditor``` instantiates ```ConstraintValueEditorHelper``` in its constructor and, at this point in time, ```dropDownData``` is guaranteed to always be ```null```. It is only when ```ConstraintValueEditor.refresh()``` is called (after ```ConstraintValueEditorHelper``` has been instantiated) that the dropdown data is actually loaded from the DMO.

IMO we should probably also look at correcting the initialisation order in ```ConstraintValueEditor```.

Comment (jomarko) [reply drools-wb-screens/drools-wb-guided-rule-editor/drools-wb-guided-rule-editor-client/src/main/java/org/drools/workbench/screens/guided/rule/client/editor/util/ConstraintValueEditorHelper.java]: @manstis I added commit according your suggestion.

**Comment by jomarko** [comment]
jenkins execute full downstream build

File: drools-wb-screens/drools-wb-guided-rule-editor/drools-wb-guided-rule-editor-client/src/main/java/org/drools/workbench/screens/guided/rule/client/editor/util/ConstraintValueEditorHelper.java [hunk 2]
```diff
@@ -239,7 +240,12 @@ private void isLHSFieldTypeEquivalent(final String boundVariable,
...
+        if (dropDownData != null) {
+            return Arrays.equals(values, dropDownData.getFixedList());
+        } else {
+            // droopDownData is null, so check if also values are null
```
Comment (manstis) [inline drools-wb-screens/drools-wb-guided-rule-editor/drools-wb-guided-rule-editor-client/src/main/java/org/drools/workbench/screens/guided/rule/client/editor/util/ConstraintValueEditorHelper.java]: Spelling: "dropDownData" not "droopDownData".

File: drools-wb-screens/drools-wb-guided-rule-editor/drools-wb-guided-rule-editor-client/src/main/java/org/drools/workbench/screens/guided/rule/client/editor/ConstraintValueEditor.java [hunk 3]
```diff
@@ -131,15 +131,18 @@ public ConstraintValueEditor(BaseSingleFieldConstraint con,
 
         setUpConstraint();
 
+        // DROOLS-2662
+        // Call refresh() before ConstraintValueEditorHelper is created
+        // This ensures dropDownData is not null
+        refresh();
```
[the remark this answers is not in the document]
Comment (manstis) [reply drools-wb-screens/drools-wb-guided-rule-editor/drools-wb-guided-rule-editor-client/src/main/java/org/drools/workbench/screens/guided/rule/client/editor/ConstraintValueEditor.java]: @Rikkola @jomarko ```Mockito.InOrder()```

**Comment by jomarko** [comment]
Just for record, the reason why `dropDownData` was null for `java.time.LocalDate` field is that `AsyncPackageDataModelOracleImpl` assumes `java.time.LocalDate` fields have type `Comparable` while `java.util.Date` fields have type `Date`. 

This difference caused that different branches of code were executed for different field types and we didn't spot the issue sooner.

If we decide to support `java.time.LocalDate` we will have to  update [converter](https://github.com/kiegroup/kie-wb-common/blob/master/kie-wb-common-services/kie-wb-common-datamodel/kie-wb-common-datamodel-backend/src/main/java/org/kie/workbench/common/services/datamodel/backend/server/builder/projects/JavaTypeSystemTranslator.java)

**Comment by jomarko** [comment]
@manstis this can be merged

**Pull request merged**